src/io/dsv.rs

```diff
@@ -30,10 +29,13 @@
 //! let table_reader: Box<dyn TableReader> = Box::new(csv_reader);
 //! ```
 //!
+//! The example first creates a [`DSVReader`] which uses comma as its separator.
+//! Then it packs it as a trait object of type [`TableReader`], which represents the visibility for the phyical layer.
```

Comment:
`represents the visibility for the physical layer`: what does this even mean? This focuses too much on operational details (“packs”) and obscures why things are happening.
```markdown
It passes the reader to the physical layer as a [`TableReader`] trait object.
```
Is easier to understand, contains less superfluous language, and still conveys the important information.